File: src/main/java/com/cdefgah/poetica/model/Answer.java
package com.cdefgah.poetica.model;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "Answers")
public final class Answer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long teamId;

    @Column(nullable = false)
    private Long questionId;

    @Column(length = 1024, nullable = false)
    private String body;

    @Column(length = 256, nullable = true)
    private String comment;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getTeamId() {
        return teamId;
    }

    public void setTeamId(Long teamId) {
        this.teamId = teamId;
    }

    public Long getQuestionId() {
        return questionId;
    }

    public void setQuestionId(Long questionId) {
        this.questionId = questionId;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Answer answer = (Answer) o;
        return id.equals(answer.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src/main/frontend/src/app/model/Answer.ts
/**
 * Модель данных ответа на вопрос (бескрылку).
 */
export class Answer {
  questionId: number;
  roundNumber: number;

  /**
   * Уникальный номер вопроса (бескрылки).
   */
  questionNumber: string = "";

  /**
   * Содержание ответа.
   */
  body: string = "";

  /**
   * Необязательный комментарий к ответу.
   */
  comment: string = "";

  /**
   * Конструктор класса.
   * @param questionNumber уникальный номер вопроса (бескрылки).
   * @param body содержимое ответа.
   * @param comment необязательный комментарий.
   */
  constructor(questionNumber: string, body: string, comment: string) {
    this.questionNumber = questionNumber;
    this.body = body;
    this.comment = comment;
  }
}
